model/device.go

```diff
@@ -200,6 +201,45 @@ func WithActuators(acts []ActuatorV2) DeviceOption {
 	}
 }
 
+// WithType sets the given type of the new device. If no type is passed, or the type is not a known type
+// this will create a device without any default settings (only MAC and name), and an empty device type.
+func WithType(deviceType string) DeviceOption {
+	return func(ctx context.Context, store datastore.Store, dev *Device) error {
```

Comment:
Why do these require context.Context and datastore.Store ? Generally with functional options you're just taking the thing you're applying the options to, i.e. Device.

Reply:
This was to allow the options to create variables, sensors and actuators. Without this, these entities would have to be passed through the device somehow.

Reply:
That's why I think we should have a ```*System``` struct instead that can hold everything together.

Reply:
Yeah, I agree. I am in the process of making factories for sensors, variables, and actuators within model, which can be exported and used in a system layer

Reply:
I'm not sure there's much point doing this though; we know what direction we should going in with this, so why don't we just start with that ?

Reply:
Sorry if that was unclear, I am implementing the new model you suggested, as a starting point, I am going to make the factories for the sensors at the model layer (#207). These factories will be used in the system layer:
```golang
func defaultControllerSensors() []*SensorV2 {
  return []SensorV2{
    BateryVoltageSensor(defaultVoltageScaleFactor),
    AnalogSensor(),
    ...
  }
}

func NewDefaultControllerSystem() *ControllerSystem {
  return NewController(mac, name, 
    WithSensors(defaultControllerSensors),
    WithActuators(defaultControllerActurators),
  )
}
```
I will close this PR

Reply:
In retrospect, I think what Saxon is referring to as the `system` layer I was describing as the `manager` layer.

I'm okay with the `system` nomenclature, though. It may be worthy of its own package that logically sits above the `model` layer.